client/my-sites/themes/theme-upload/index.jsx

```diff
@@ -213,7 +222,7 @@ class Upload extends React.Component {
 			<UpsellNudge
 				title={ translate( 'Upgrade to the Business plan to access the theme install features' ) }
 				event="calypso_theme_install_upgrade_click"
-				href={ `/checkout/${ siteId }/business?redirect_to=/themes/upload/${ siteId }` }
+				href={ `/checkout/${ siteId }/business?redirect_to=/themes/upload/${ siteId }?notice=purchase-success` }
```

Comment:
This double query string (`?redirect_to=...?notice=...`) makes me nervous but I guess it's probably safe if it works. This is similar to some work we've just been considering regarding encoding of the redirect URL. I wonder if we could encode this `redirect_to` param and then decode it before the redirect. 🤔  Anyway, as I said, it's probably safe here if it works for you.

Reply:
As you know from our Slack convo, I went ahead and encoded the redirect param - Checkout seems smart enough to parse that. If only I could thank the Checkout architect…